source/Audio.cpp

```diff
@@ -260,6 +267,55 @@ void Audio::Update(const Point &listenerPosition)
 
 
 
+void Audio::UpdateMusic(PlayerInfo &player, Track::GameState state)
+{
+	if(!isInitialized)
+		return;
+	
+	if(Preferences::Has("ingame music"))
+	{
+		bool currentPlaylistValid = currentPlaylist ? currentPlaylist->MatchingConditions(player) : false;
+		if(currentTrack->IsFinished() || !currentPlaylistValid)
+		{
+			if(!currentPlaylistValid)
```

Comment:
mb add a comment we look for a new valid playlist in this case
what if the track is finished? shouldnt we reroll it?

Reply:
depends on the progression method, look further down

Reply:
Definitely could use a few comments at some point